src/components/ratings-and-reviews-components/review-and-rating-post-form/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { SyntheticEvent, useEffect } from 'react';
+import { Review } from '@services/generated-api/data-contracts';
+import api from '@services/api';
+import { useFormAndValidation } from '@hooks/use-form-and-validation';
+import RatingInput from '@components/ratings-and-reviews-components/rating-input';
+import Button from '@components/Button';
+import styles from './review-and-rating-post-form.module.scss';
+
+const dateOptions = { day: '2-digit', month: '2-digit', year: '2-digit' };
+
+interface IReviewAndRatingPostForm {
+	review: Review | null;
+	productId: number;
+	onAddReview: (review: Review) => void;
+	onUpdateReview: (review: Review) => void;
+}
+
+const ReviewAndRatingPostForm: React.FC<IReviewAndRatingPostForm> = ({
+	review,
+	productId,
+	onAddReview,
+	onUpdateReview,
+}) => {
+	const { values, setValues, handleChange } = useFormAndValidation({
+		text: review?.text || '',
+		score: review?.score || 0,
+	});
+
+	useEffect(() => {
+		review && setValues({ text: review?.text, score: review?.score });
+	}, [review, productId, setValues]);
+
+	const handleReviewSubmit = (e: SyntheticEvent) => {
+		e.preventDefault();
+		// TODO: Add error processing in catch block
+		if (review?.score && values.text && values.text !== review?.text) {
+			api
+				.reviewsUpdate(productId, review.id, { text: values.text as string })
+				.then((res) => onUpdateReview(res))
+				.catch((err) => console.log(err));
```

Comment:
Логику обработок ошибок лучше выполнять в слое API, например:

```suggestion
reviewsUpdate(productId: number, reviewId: number, data: ReviewUpdate) {
		return this._request(`products/${productId}/reviews/${reviewId}/`, {
			method: 'PATCH',
			headers: {
				'Content-Type': 'application/json',
				Authorization: `Token ${Cookies.get('token')}`,
			},
			body: JSON.stringify(data),
		}).catch(console.log);
	}
```

Reply:
@Karmyshkov а разве в этом случае все блоки then, которые я повешу на вызов этого метода из api не встанут после catch? Так же не очень хорошо?

Reply:
@kavabunga, согласен с тобой